tests/step4_if_fn_do.mal

```diff
@@ -203,6 +203,8 @@ a
 ;; Testing recursive function in environment.
 (let* [cst (fn* [n] (if (= n 0) nil (cst (- n 1))))] (cst 1))
 ;=>nil
+(let* [f (fn* [n] (if (= n 0) 0 (g (- n 1)))) g (fn* [n] (f n))] (f 2))
+;=>0
 
```

Comment:
Vectors are deferrable until step A, so either this test and the previous one should be deferrable, or (better) the vectors should be replaced with lists.